students/20th/doeunkim/feed.js

```diff
@@ -0,0 +1,295 @@
+}
+
+function sublike() {
+    likeBtn.classList.toggle("pop");
+    red.style.display = "none";
+    likes.innerHTML = "좋아요 19개";
+}
+
+likeBtn.addEventListener("click", addlike);
+red.addEventListener("click", sublike);
+
+
+
+//아이디 검색
+
+const typing = document.querySelector(".search input");
+const searchList = document.querySelector(".searchList");
+
+const usersToSearch = [{
+    id: 'skuukzky',
+    profileImg: 'img/skuukzky.jpg',
+    description: '숮이 💄💅👡👠🎀👙🌂👗🌂🎀💋💌'
+}, {
+    id: '_imyour_joy',
+    profileImg: 'img/_imyour_joy.jpg',
+    description: 'Joy 🌺'
+}, {
+    id: 'cafeknotted',
+    profileImg: 'img/cafeknotted.jpg',
+    description: '노티드 Cafe Knotted'
+}, {
+    id: 'jennierubyjane',
+    profileImg: 'img/jennierubyjane.jpg',
+    description: 'J'
+}, {
+    id: 'bts.bighotofficial',
+    profileImg: 'img/bts.bighotofficial.jpg',
+    description: 'BTS | BANGTAN | 방탄소년단'
+}, {
+    id: 'skawngur',
+    profileImg: 'img/skawngur.jpg',
+    description: '남주혁'
+}, {
+    id: '_victoria_bakery',
+    profileImg: 'img/_victoria_bakery.jpg',
+    description: 'Victoria Bakery'
+}, {
+    id: 'yeojin9oo',
+    profileImg: 'img/yeojin9oo.jpg',
+    description: '여진구 Yeo Jin Goo'
+}, {
+    id: 'akmu_suhyun',
+    profileImg: 'img/akmu_suhyun.jpg',
+    description: '이수현 1999.05.04'
+}, {
+    id: 'nct_dream',
+    profileImg: 'img/nct_dream.jpg',
+    description: 'NCT DREAM OFFICIAL'
+}, {
+    id: 'wm_ohmygirl',
+    profileImg: 'img/wm_ohmygirl.jpg',
+    description: 'OHMYGIRL (오마이걸) WM ENT 오마이걸 공식 인스타그램'
+}, {
+    id: 'twicetagram',
+    profileImg: 'img/twicetagram.jpg',
+    description: 'TWICE'
+}, {
+    id: 'youngji_02',
+    profileImg: 'img/youngji_02.jpg',
+    description: '이영지'
+}, {
+    id: 'saddlerhaus_bakery',
+    profileImg: 'img/saddlerhaus_bakery.jpg',
+    description: 'french waffle'
+}, {
+    id: 'onyourm__ark',
+    profileImg: 'img/onyourm__ark.jpg',
+    description: 'Mark 🐯🌱'
+}, {
+    id: 'songkang_b',
+    profileImg: 'img/songkang_b.jpg',
+    description: '송강'
+}, {
+    id: 'roses_are_rosie',
+    profileImg: 'img/roses_are_rosie.jpg',
+    description: 'ROSÉ'
+}, {
+    id: 'dlwlrma',
+    profileImg: 'img/dlwlrma.jpg',
+    description: '이지금 IU'
+}];
+
+function showAllId() {
+    searchList.innerHTML="";
+    usersToSearch.forEach(i => {
+        const matchedId = document.createElement('div');
+        matchedId.innerHTML = `<div class="searchedUser">
+        <img alt="user's profile image" src=${i.profileImg}>
+        <div class="userId">
+        <p class="id">${i.id}</p>
+        <p class="gray twelve" id="description">${i.description}</p>
+        </div>
+        </div>`
+        searchList.appendChild(matchedId);
+    }
+    )}
+    
+    function showSomeId() {
+        searchList.innerHTML="";
+    const matchedUsers = [];
+    usersToSearch.forEach((i) => {  
+    i.id.startsWith(typing.value)? matchedUsers.push(i) : null;
+    });
```

Comment:
```suggestion
    function showSomeId() {
      searchList.innerHTML="";
      const matchedUsers = [];
      
      USERS.forEach(user => {  
        user.id.startsWith(typing.value) && matchedUsers.push(user);
      });
```
- indent 잘 지켜가며 코드를 작성하셔야 코드를 읽기 쉽고, 문제가 생긴 부분을 찾기도 쉽습니다!
- 로직이 달성하고자 하는 바를 다시 생각해보면 `전체 유저 중에 조건을 만족하는 사용자로 이루어진 새로운 배열을 리턴`하는 것입니다. 그렇다면 전체 유저 중에 일부를 걸러내는 것이고, `filter`를 사용할 수 있습니다.
```js
const matchedUsers = USERS.filter(user => user.id.starts.With(typing.value))
```